appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -41,204 +41,234 @@
+                return validateTableConnectionValidation(jdbcPool);
+            case CUSTOM_VALIDATION_CLASS_NAME_MANDATORY:
+                return validateClassConnectionValidation(jdbcPool);
+            case RES_TYPE_MANDATORY:
+                return validateResourceType(jdbcPool);
+            default:
+                return true;
+        }
+    }
 
-        if (variableReference.length == 1) {
-            //We got a system variable as no split occured
-            return System.getProperty(variableReference[0]);
+
+    private boolean validateSteadyPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
+        return steadyPoolSize != null && steadyPoolSize >= 0;
+    }
+
+
+    private boolean validateMaxPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
+        final String propertyValue = jdbcPool.getMaxPoolSize();
+        final String value = propertyValue == null ? Constants.DEFAULT_MAX_POOL_SIZE : propertyValue;
+        final Integer maxPoolSize = toInteger(resolve(value));
+        if (maxPoolSize == null || maxPoolSize < 1) {
+            return false;
+        }
+        if (steadyPoolSize != null && steadyPoolSize > maxPoolSize) {
+            return false;
         }
+        return true;
+    }
 
-        String variableToFind = variableReference[1];
 
-        switch (variableReference[0]) {
-            case "ENV":
-                //Check environment variables for requested value
-                String varValue = System.getenv(variableToFind);
-                if (varValue != null && !varValue.isEmpty()) {
-                    return varValue;
-                }
-                break;
-            case "MPCONFIG":
-                //Check microprofile config for requested value
-                Config config = ConfigProvider.getConfig();
-                varValue = config.getValue(variableToFind, String.class);
-                if (varValue != null && !varValue.isEmpty()) {
-                    return varValue;
-                }
-                break;
+    private boolean validateWrapping(final JdbcConnectionPool jdbcPool) {
+        final String stmtCacheSize = jdbcPool.getStatementCacheSize();
+        final String stmtLeakTimeout = jdbcPool.getStatementLeakTimeoutInSeconds();
+        final boolean wrappingEnabled = isTrue(jdbcPool.getWrapJdbcObjects());
+        if (!wrappingEnabled && !isEmpty(jdbcPool.getSqlTraceListeners())) {
+            return false;
+        }
+        if (!wrappingEnabled && isPositiveInt(stmtCacheSize)) {
+            return false;
         }
-        
-        //If this point is reached, the variable value could not be found
-        return null; 
+        if (!wrappingEnabled && isPositiveInt(stmtLeakTimeout)) {
+            return false;
+        }
+        if (!wrappingEnabled && isTrue(jdbcPool.getStatementLeakReclaim())) {
```

Comment:
If can be replaced by single return

Reply:
It can, but it would be harder to read. Reading conditions on more than single line is harder than two if conditions with implicit else.
If the first IF is resolved to false, then it does not return anything and evaluates the second IF. 

Compiler optimization or at least JIT optimizer can do that automatically and even if it would not do that, this code still may be faster than joined evaluation (but one nanosecond is not worth to analyze that, we have hundred of miliseconds in other places)..